src/runtime/GlobalObjectBuiltinArray.cpp

```diff
@@ -1498,6 +1543,42 @@ static Value builtinArrayPush(ExecutionState& state, Value thisValue, size_t arg
     return Value(n);
 }
 
+// https://www.ecma-international.org/ecma-262/10.0/#sec-array.prototype.flat
+// Array.prototype.flat( [ depth ] )
+static Value builtinArrayFlat(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    RESOLVE_THIS_BINDING_TO_OBJECT(O, Array, flat);
+    int64_t sourceLen = O->lengthES6(state);
+    double depthNum = 1;
+    if (argc > 0) {
+        depthNum = argv[0].toInteger(state);
+    }
+    Object* A = arraySpeciesCreate(state, O, 0);
+    Value args[] = { O, Value(sourceLen), Value(0), Value(depthNum) };
+    flattenIntoArray(state, A, 4, args, true);
+
+    return A;
+}
+
+// https://www.ecma-international.org/ecma-262/10.0/#sec-array.prototype.flatmap
+// Array.prototype.flatMap ( mapperFunction [ , thisArg ] )
+static Value builtinArrayFlatMap(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    RESOLVE_THIS_BINDING_TO_OBJECT(O, Array, flat);
+    int64_t sourceLen = O->lengthES6(state);
+    if (!argv[0].isCallable()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, state.context()->staticStrings().Array.string(), true, state.context()->staticStrings().find.string(), errorMessage_GlobalObject_CallbackNotCallable);
```

Comment:
Some arguments for `throwBuiltinError` are incorrect. 
`state.context()->staticStrings().find.string()` as `state.context()->staticStrings().flatMap.string()`
`errorMessage_GlobalObject_CallbackNotCallable` as `errorMessage_GlobalObject_FirstArgumentNotCallable`